src/deferred-promise.ts

```diff
@@ -0,0 +1,60 @@
+type DeferredPromiseOptions<T> = {
+    timeout?: number;
+    onTimeout?: (resolve: (value: T) => void, reject: (reason: Error) => void) => void;
+};
+
+/** Creates a promise and exposes its resolve and reject methods, with an optional timeout. */
+export class DeferredPromise<T> extends Promise<T> {
+    resolve!: (value: T) => void;
+    reject!: (reason: unknown) => void;
```

Comment:
Do we need the assertions here?

```suggestion
    resolve: (value: T) => void;
    reject: (reason: unknown) => void;
```

Reply:
yeah, this is more of a late intialization assertion. typescript isn't smart enough to understand they're being set in the constructor

Reply:
Not sure why though - when testing locally, I don't see it complaining when I remove the assertions - in what situations are you seeing errors?

Reply:
ah, the structure seems to have changed back to a different setup, this is what I was thinking about: [`25d9059` (#161)](https://github.com/mongodb-js/mongodb-mcp-server/pull/161/commits/25d905907f738591e7fc1d8d22708ca513c9829a) (cleaner to do this imo)

Reply:
okay I'm realizing now the assertions were leftover from an older implementation that had a different structure; removed them now. as we assert them later on instead.